pinns/src/pinns.c

```diff
@@ -196,25 +208,29 @@ int main(int argc, char **argv) {
         pause();
       _exit (EXIT_SUCCESS);
     }
+
     if (TEMP_FAILURE_RETRY(close(p[1])) < 0)
       pexit("Failed to close pipe");
 
-    /* Wait for user namespace creation.  */
-    res = '1';
-    if (TEMP_FAILURE_RETRY(read(p[0], &res, 1)) < 0 || res != '0')
-      pexit("Failed to read from the sync pipe");
+    if (bind_user) {
+      /* Wait for user namespace creation.  */
+      res = '1';
+      if (TEMP_FAILURE_RETRY(read(p[0], &res, 1)) < 0 || res != '0')
+        pexit("Failed to read from the sync pipe");
 
-    if (gid_mapping && write_mapping_file(pid, gid_mapping, true) < 0)
-      pexit("Cannot write gid mappings");
+      /* Write user mappings */
+      if (gid_mapping && write_mapping_file(pid, gid_mapping, true) < 0)
+        pexit("Cannot write gid mappings");
 
-    if (uid_mapping && write_mapping_file(pid, uid_mapping, false) < 0)
-      pexit("Cannot write gid mappings");
+      if (uid_mapping && write_mapping_file(pid, uid_mapping, false) < 0)
```

Comment:
nit: there seem to be some whitespace changes I wouldnt' expect (not sure if they're intentional)

Reply:
Intentional: These were indented into the `if (bind_user) {` on [line 215 above](https://github.com/cri-o/cri-o/pull/4508/files/60986224f0d5dd4879aef283a5713e308e0533bd#diff-bfef5beff459416de7e125890a18464df27f7ada77aa3d6f89eb033fac790801R215).